examples/ssr_router/Cargo.toml

```diff
@@ -0,0 +1,24 @@
+[package]
+name = "ssr_router"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+yew = { path = "../../packages/yew", features = ["ssr", "hydration", "trace_hydration"] }
+function_router = { path = "../function_router" }
+log = "0.4"
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
```

Comment:
> Do we want to use target_arch = "wasm32" as our switch to differentiate between server and client?

No it doesn't. This is to select the available runtime.

Both csr and ssr works in Wasm environment. This is how the tests are written.

Reply:
This about the example though, which will be copy-pasted and reused a lot more often, hence why I think it's more critical here than in usual test code.

Reply:
> the ServerApp component there is only exposed on non-wasm target, which is a further example of what I mean with the question.

Clippy wouldn't pass if you don't exclude the server app as it's binary (bin).

Reply:
Can you split the example into a `lib` and a `bin`? Then clippy shouldn't complain if both types are `pub`.

Reply:
ServerApp is provided by the `function_router` example, not the `ssr_router` example which in itself is an example that is both a bin and a lib.

I am not a fan of adding a crate in the examples folder that does not actually do anything as that's more confusing.

Reply:
Have a look at https://github.com/WorldSEnder/yew/commit/9391809de8f750f9c31d6d011cd95981702767d8 for how I thought this might look in the `function_router` example. Incidentally, this already is both a lib and a bin crate, I didn't realize that before. Both examples build fine for me there. Although: I just noticed that the `simple_ssr` example flashes a uuid from the server before it gets replaced with a freshly fetched one. During code review, I assumed it would suspend and render the "Loading..." fallback for some reason.